internal/storage/block_storage.go

```diff
@@ -532,3 +534,32 @@ func (b *BlockStorage) FindTransaction(
 		newestBlock.Index,
 	)
 }
+
+// AtTip returns a boolean indicating if we
+// are at tip (provided some acceptable
+// tip delay).
+func (b *BlockStorage) AtTip(
+	ctx context.Context,
+	tipDelay int64,
+) (bool, error) {
```

Comment:
nit: do we want to return just error not both bool and error?

Reply:
In the places where I call this function, I found it more aesthetic to look at the error only if we could not tell if we were at tip. Very much my opinion and your suggestion would also work (if we did an `errors.Is`):

Example:
```
	atTip, err := b.helper.AtTip(ctx, b.tipDelay)
	if err != nil {
		return fmt.Errorf("%w: unable to determine if at tip", err)
	}

	if (!atTip && !b.broadcastBehindTip) && onlyEligible {
		return nil
	}

```